File: Year-2023/LeetCode-March-Challenge/CheckCompletenessofaBinaryTree.java
import java.util.*;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode() {
    }

    TreeNode(int val) {
        this.val = val;
    }

    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public boolean isCompleteTree(TreeNode root) {
        if (root == null)
            return true;
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);
        boolean check = false;
        while (!q.isEmpty()) {
            TreeNode curr = q.poll();
            if (curr.left != null) {
                if (check)
                    return false;
                q.add(curr.left);
            } else
                check = true;
            if (curr.right != null) {
                if (check)
                    return false;
                q.add(curr.right);
            } else
                check = true;
        }
        return true;
    }
}

public class CheckCompletenessofaBinaryTree {
    public static void main(String[] args) {
        TreeNode p1 = new TreeNode(1);
        TreeNode p2 = new TreeNode(2);
        TreeNode p3 = new TreeNode(3);
        TreeNode p4 = new TreeNode(4);
        TreeNode p5 = new TreeNode(5);
        TreeNode p6 = new TreeNode(6);

        p1.left = p2;
        p1.right = p3;
        p2.left = p4;
        p2.right = p5;
        p3.left = p6;

        Solution s = new Solution();
        System.out.println(s.isCompleteTree(p1));
    }
}

File: Year-2023/LeetCode-March-Challenge/CheckCompletenessofaBinaryTree.js
function TreeNode(val, left, right) {
    this.val = (val === undefined ? 0 : val)
    this.left = (left === undefined ? null : left)
    this.right = (right === undefined ? null : right)
}

var isCompleteTree = function (root) {
    if (!root) return true;
    let q = [root];
    let check = false;

    while (q.length) {
        let curr = q.shift();
        if (curr.left) {
            if (check) return false;
            q.push(curr.left);
        } else check = true;

        if (curr.right) {
            if (check) return false;
            q.push(curr.right);
        } else check = true;
    }
    return true;
};

let p1 = new TreeNode(1);
let p2 = new TreeNode(2);
let p3 = new TreeNode(3);
let p4 = new TreeNode(4);
let p5 = new TreeNode(5);
let p6 = new TreeNode(6);

p1.left = p2;
p1.right = p3;
p2.left = p4;
p2.right = p5;
p3.left = p6;

console.log(isCompleteTree(p1));